pandas/plotting/_matplotlib/hist.py

```diff
@@ -61,7 +61,8 @@ def _args_adjust(self):
         # where subplots are created based on by argument
         if is_integer(self.bins):
             if self.by is not None:
-                grouped = self.data.groupby(self.by)[self.columns]
+                bymodi = fix_groupby_singlelist_input(self.by)
+                grouped = self.data.groupby(bymodi)[self.columns]
```

Comment:
Why is this necessary?

Reply:
because `.hist` and `.box` use groupby internally in a single way. For example if I did hist by ['a','b','c','d'] the results will be like (a,), (b,), (c,), (d,).
some plotting functions and the pivot table are actually iterating over groupby.

Reply:
But here, `grouped` is only being used in L66 immediately below, right?

    self.bins = [self._calculate_bins(group) for key, group in grouped]

In this usage, only the group is being used and the key is ignored. So why is this needed if it's only the key changing?